src/ert/gui/model/job_list.py

```diff
@@ -129,65 +126,51 @@ def parent(self, child: Optional[QModelIndex] = None) -> Optional[QObject]:
+                if real_index.isValid() and sm.hasChildren(real_index):
+                    return sm.index(proxyIndex.row(), proxyIndex.column(), real_index)
+        return QModelIndex()
 
-    def mapFromSource(self, sourceIndex: QModelIndex) -> QModelIndex:
-        if not sourceIndex.isValid():
-            return QModelIndex()
-        if not self._accept_index(sourceIndex):
-            return QModelIndex()
-        return self.index(sourceIndex.row(), sourceIndex.column(), QModelIndex())
+    def mapFromSource(self, src_index: QModelIndex) -> QModelIndex:
+        if src_index.isValid():
+            source_node = src_index.internalPointer()
+
+            if source_node and self._accept_index(src_index):
+                return self.index(src_index.row(), src_index.column(), QModelIndex())
+        return QModelIndex()
 
     def _source_data_changed(
         self, top_left: QModelIndex, bottom_right: QModelIndex, roles: List[int]
     ) -> None:
-        if not self._accept_index(top_left):
-            return
-        proxy_top_left = self.mapFromSource(top_left)
-        proxy_bottom_right = self.mapFromSource(bottom_right)
-        if not proxy_top_left.isValid() or not proxy_bottom_right.isValid():
-            return
-        self.dataChanged.emit(proxy_top_left, proxy_bottom_right, roles)
+        if self._accept_index(top_left):
+            proxy_top_left = self.mapFromSource(top_left)
+            proxy_bottom_right = self.mapFromSource(bottom_right)
+            if all([proxy_top_left.isValid(), proxy_bottom_right.isValid()]):
+                self.dataChanged.emit(proxy_top_left, proxy_bottom_right, roles)
 
     def _accept_index(self, index: QModelIndex) -> bool:
-        if index.internalPointer() is None:
-            return False
-
-        # This model should only consist of job indices, so anything else mean
-        # the index is not on "our branch" of the state graph.
-        if not index.data(IsJobRole):
-            return False
-
-        # traverse upwards and check real and iter against parents of
-        # this index.
-        while index.isValid() and index.internalPointer() is not None:
-            if (index.data(IsRealizationRole) and (index.row() != self._real)) or (
-                index.data(IsEnsembleRole) and (index.row() != self._iter)
-            ):
-                return False
-            index = index.parent()
-        return True
+        if index.internalPointer() and index.data(IsJobRole):
+            # traverse upwards and check real and iter against parents of this index
+            while index.isValid() and index.internalPointer():
+                if (index.data(IsRealizationRole) and (index.row() != self._real)) or (
+                    index.data(IsEnsembleRole) and (index.row() != self._iter)
+                ):
+                    return False
+                index = index.parent()
+            return True
+        return False
```

Comment:
I would prefer it with less nesting.
```
if not index.internalPointer() or not index.data(IsJobRole):
    return False
```
at the the top and the rest the same. but up to you. I will approve it this way too

Reply:
Let's go with the less nesting part. It's easier to read. 👍